frontend/src/scenes/settings/project/SessionRecordingSettings.tsx

```diff
@@ -217,6 +227,99 @@ export function ReplayAuthorizedDomains(): JSX.Element {
     )
 }
 
+function variantOptions(multivariate: MultivariateFlagOptions | undefined): LemonSegmentedButtonOption<string>[] {
+    if (!multivariate) {
+        return []
+    }
+    return [
+        {
+            label: 'any',
+            value: 'any',
+        },
+        ...multivariate.variants.map((variant) => {
+            return {
+                label: variant.key,
+                value: variant.key,
+            }
+        }),
+    ]
+}
+
+function LinkedFlagSelector(): JSX.Element | null {
+    const { updateCurrentTeam } = useActions(teamLogic)
+    const { currentTeam } = useValues(teamLogic)
+
+    const { hasAvailableFeature } = useValues(userLogic)
+    const { featureFlags } = useValues(featureFlagLogic)
+    const flagIsEnabled = featureFlags[FEATURE_FLAGS.SESSION_RECORDING_SAMPLING]
+    const featureFlagRecordingFeatureEnabled =
+        flagIsEnabled || hasAvailableFeature(AvailableFeature.REPLAY_FEATURE_FLAG_BASED_RECORDING)
+
+    const [selectedFlag, setSelectedFlag] = useState<FeatureFlagBasicType | null>(null)
+
+    if (!featureFlagRecordingFeatureEnabled) {
+        return null
+    }
+
+    return (
+        <>
+            <div className="flex flex-col space-y-2">
+                <LemonLabel className="text-base">Enable recordings using feature flag</LemonLabel>
+                <p>Linking a flag means that recordings will only be collected for users who have the flag enabled.</p>
+                <div className="flex flex-row justify-start">
+                    <FlagSelector
+                        value={currentTeam?.session_recording_linked_flag?.id ?? undefined}
+                        onChange={(id, key, flag) => {
+                            setSelectedFlag(flag)
+                            updateCurrentTeam({ session_recording_linked_flag: { id, key } })
```

Comment:
Should we be resetting the variant here in case you move from a multivariant flag to a boolean flag? Otherwise you might end up with a `variant` that points to a non-existent variant for the current flag
```suggestion
                            updateCurrentTeam({ session_recording_linked_flag: { id, key, variant: null } })
```

Reply:
Ah, damn, was sure I'd applied this 🙃